plugins/modules/inventory_intent.py

```diff
@@ -2062,27 +2113,43 @@ def check_device_role(self, device_ip):
 
         return response.get('role') == role and response.get('roleSource') == role_source
 
-    def check_interface_details(self, device_ip):
+    def check_interface_details(self, device_ip, interface_name):
         """
-        Checks if the interface details for a device in Cisco DNA Center match the specified values in the configuration.
-        Args:
-            self (object): An instance of a class used for interacting with Cisco DNA Center.
+        Checks if the interface details for a device in Cisco Catalyst Center match the specified values in the configuration.
+        Parameters:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
             device_ip (str): The management IP address of the device for which interface details are to be checked.
         Returns:
             bool: True if the interface details match the specified values, False otherwise.
         Description:
-            This method retrieves the interface details for a device in Cisco DNA Center using the 'get_interface_by_ip' API call.
+            This method retrieves the interface details for a device in Cisco Catalyst Center using the 'get_interface_by_ip' API call.
             It then compares the retrieved details with the specified values in the configuration for updating interface details.
             If all specified parameters match the retrieved values or are not provided in the playbook parameters, the function
             returns True, indicating successful validation.
         """
+        device_id = self.get_device_ids([device_ip])
+
+        if not device_id:
+            self.log("""Error: Device with IP '{0}' not found in Cisco Catalyst Center.Unable to update interface details."""
+                     .format(device_ip), "ERROR")
+            return False
 
+        interface_detail_params = {
+            'device_id': device_id[0],
+            'name': interface_name
+        }
         response = self.dnac._exec(
             family="devices",
-            function='get_interface_by_ip',
-            params={"ip_address": device_ip}
+            function='get_interface_details',
+            params=interface_detail_params
         )
-        response = response.get("response")[0]
+        self.log("Received API response from 'get_interface_details': {0}".format(str(response)), "DEBUG")
+        response = response.get("response")
+
+        if not response:
+            self.log("Doesnot get the response of 'get_interface_details' api.", "DEBUG")
```

Comment:
self.log("No response received from the API 'get_interface_details'.", "DEBUG")

Reply:
added.